log-appender/src/main/scala/io/phdata/pulse/Stream.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2018 phData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.phdata.pulse
+
+import java.nio.charset.Charset
+import java.util.concurrent.TimeUnit
+
+import com.typesafe.scalalogging.LazyLogging
+import io.phdata.pulse.log.JsonParser
+import monix.reactive.subjects.ConcurrentSubject
+import monix.execution.Scheduler.Implicits.global
+import org.apache.http.client.methods.HttpPost
+import org.apache.http.entity.StringEntity
+import org.apache.http.impl.client.{ CloseableHttpClient, HttpClientBuilder }
+import org.apache.log4j.spi.LoggingEvent
+
+import scala.collection.mutable
+import scala.concurrent.duration.{ Duration, FiniteDuration }
+
+trait Stream[E] extends LazyLogging {
+  var batchFlushDuration: FiniteDuration = Duration(1, TimeUnit.SECONDS)
+  var batchFlushSize: Int                = 0
+
+  // Create a subject we can write events to
+  val subject = ConcurrentSubject.publish[E]
+
+  subject
+    .bufferTimedAndCounted(batchFlushDuration, batchFlushSize)
+    .foreach(y => save(y))
+
+  def append(value: E): Unit = subject.onNext(value)
+
+  def save(values: Seq[E])
+}
+
+class HttpStream(flushDuration: Int, flushSize: Int) extends Stream[LoggingEvent] {
+  this.batchFlushDuration = Duration(flushDuration, TimeUnit.SECONDS)
+  this.batchFlushSize = flushSize
+
+  //Create Closeable HttpClient to save http events
+  val client: CloseableHttpClient = HttpClientBuilder.create().build();
+
+  override def save(values: Seq[LoggingEvent]): Unit = {
+    val logArray   = values.toArray
+    val jsonParser = new JsonParser
+    val logMessage = jsonParser.marshallArray(logArray)
+
+    val post: HttpPost = new HttpPost("http://edge2.valhalla.phdata.io")
+    post.setHeader("Content-type", "application/json")
+    post.setEntity(new StringEntity(logMessage, Charset.forName("UTF8")))
+    logger.debug("Executing request:" + post.getRequestLine)
+    try {
+      val response   = client.execute(post)
+      val statusCode = response.getStatusLine.getStatusCode
+      response.close()
+      logger.debug("Response code: " + statusCode)
```

Comment:
We can't use logging here, because we don't want it to show up in the user logs or create an infinite loop of logs. See the HttpAppender and `LogLog.debug`